dll/win32/shimgvw/shimgvw.c

```diff
@@ -374,18 +374,21 @@ Preview_pFreeImage(PPREVIEW_DATA pData)
         GdipDisposeImage(g_pImage);
         g_pImage = NULL;
     }
-
-    pData->m_szFile[0] = UNICODE_NULL;
 }
 
 static VOID
 Preview_pLoadImage(PPREVIEW_DATA pData, LPCWSTR szOpenFileName)
 {
     HRESULT hr;
+    BOOL bCanDel;
+    HWND hToolbar = GetDlgItem(g_hMainWnd, IDC_TOOLBAR);
     Preview_pFreeImage(pData);
     InvalidateRect(pData->m_hwnd, NULL, FALSE); /* Schedule redraw in case we change to "No preview" */
 
+    GetFullPathNameW(szOpenFileName, _countof(g_szFile), g_szFile, NULL);
     hr = LoadImageFromPath(szOpenFileName, &g_pImage);
+    bCanDel = hr != HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND) && hr != HRESULT_FROM_WIN32(ERROR_PATH_NOT_FOUND);
+    SendMessageW(hToolbar, TB_ENABLEBUTTON, IDC_DELETE, bCanDel);
```

Comment:
If the hToolbar is going to be used only once in this function, then you could do instead:
```suggestion
    SendDlgItemMessageW(g_hMainWnd, IDC_TOOLBAR, TB_ENABLEBUTTON, IDC_DELETE, bCanDel);
```